server/src/modules/auth/auth.controller.ts

```diff
@@ -111,10 +111,11 @@ export class AuthController {
 
     const otp = generateRandomDigits(OTP_LENGTH)
     const hashedOtp = await hashData(otp)
+    const ip = req.header('CF-Connecting-IP') || req.ip
```

Comment:
Maybe leave commentary here explaining what this is doing, and why. Take a cue from the comments in the code from where you found this line.